packages/core/src/exchange/acceptOffer.ts

```diff
@@ -57,8 +57,15 @@ export async function acceptOffer(
 ): Promise<UUID> {
   const address = await signer.getAddress()
 
-  const { offer } = await sdk.FetchOffer({ offerId })
-  invariant(offer, "Can't find offer")
+  let offer = undefined
+  const { listing } = await sdk.FetchListing({ offerId })
+  if (listing) {
+    offer = listing
+  } else {
+    const { openOffer } = await sdk.FetchOpenOffer({ offerId })
+    invariant(openOffer, "Can't find offer")
+    offer = openOffer
+  }
```

Comment:
I would move this to the calling functions that have the context and pass the offer as parameter. This `acceptOffer` is internal and called by the `acceptBid` and `buyToken` so the parent function have the required context

Reply:
Done here [48eb35c](https://github.com/liteflow-labs/liteflow-js/pull/225/commits/48eb35cc594488751df1b76b32e199e16022c340)